odoo/models.py

```diff
@@ -67,7 +67,7 @@
 _schema = logging.getLogger(__name__ + '.schema')
 _unlink = logging.getLogger(__name__ + '.unlink')
 
-regex_order = re.compile('^(\s*([a-z0-9:_]+|"[a-z0-9:_]+")(\s+(desc|asc))?\s*(,|$))+(?<!,)$', re.I)
```

Comment:
You want to support `nulls first` too, as it's not the default for ASC ordering.
And it's possible to specify a NULLS option even without ASC/DESC.

The following expression should do that :
`r'^(\s*([a-z0-9:_]+|"[a-z0-9:_]+")(\s+(desc|asc))?(\s+nulls\s+(first|last))?\s*(,|$))+(?<!,)$'`

Besides, the string should be a raw string as a best practice (it wasn't because we overlooked it)